src/app/entity/entity-mapper.service.ts

```diff
@@ -26,18 +26,27 @@ import { Entity } from './entity';
 @Injectable()
 export class EntityMapperService {
 
+  private static getDatabaseIdByEntity<T extends Entity>(entity: T): string {
+    return EntityMapperService.getDatabaseId(entity.getType(), entity.getId());
+  }
+
+  private static getDatabaseId(type: string, id: string): string {
```

Comment:
reading through this I initially got confused. Suggesting to rename this to `generateDatabaseId()`, `computeDatabaseId()` or something similar